src/app/model/Mission.ts

```diff
@@ -168,15 +168,15 @@ class Mission extends BaseModel {
   FundedStatus = MissionFundedStatus;
   TimeWindowType = TimeWindowType;
 
-  selectInProposed = (state: any) => state.firestore.ordered.missionsInProposed || [];
+  selectInProposed = (state: any) => state.firestore?.ordered?.missionsInProposed || [];
   fsInProposed = fsInProposed;
-  selectInPlanning = (state: any) => state.firestore.ordered.missionsInPlanning || [];
+  selectInPlanning = (state: any) => state.firestore?.ordered?.missionsInPlanning || [];
   fsInPlanning = fsInPlanning;
-  selectInProgress = (state: any) => state.firestore.ordered.missionsInProgress || [];
+  selectInProgress = (state: any) => state.firestore?.ordered?.missionsInProgress || [];
   fsInProgress = fsInProgress;
-  selectInDone = (state: any) => state.firestore.ordered.missionsInDone || [];
+  selectInDone = (state: any) => state.firestore?.ordered?.missionsInDone || [];
   fsInDone = fsInDone;
-  selectIncomplete = (state: any) => state.firestore.ordered.incompleteMissions || [];
+  selectIncomplete = (state: any) => state.firestore?.ordered?.incompleteMissions || [];
```

Comment:
Does adding the ?. Checks change the implication that the state object should already have this shape? 

This model already gives us the expectation that these properties exist, right?

Reply:
I added them because I didn't mock the state for the unit test in `MissionControl.test.js` file and the test failed there. If we are 💯sure that the state will always have those values defined, then I have to mock the state in the test. What do you think?

Reply:
Due to the nature of firestore's noSQL-ness we should definitely be using optional chaining everywhere just in case. 👍